In [2]:
domains_tokens = {
    "CC": 2281,
    "Stack": 411,
    "C4": 198,
    "Reddit": 89,
    "Pes20": 70,
    "Gutenberg": 6,
    "Wikipedia": 4.3
}

all_tokens = sum(domains_tokens.values())

for domain, tokens in domains_tokens.items():
    print(f"{domain}: {tokens/all_tokens:.2%}")

CC: 74.56%
Stack: 13.43%
C4: 6.47%
Reddit: 2.91%
Pes20: 2.29%
Gutenberg: 0.20%
Wikipedia: 0.14%


In [3]:
# only take some domains but keep the relative proportions
selected_domains = {
    "CC": domains_tokens["CC"],
    "Books": domains_tokens["Gutenberg"],
    "Stack": domains_tokens["Stack"],
    "Pes20": domains_tokens["Pes20"],
    "Reddit": domains_tokens["Reddit"]
}

selected_tokens = sum(selected_domains.values())

for domain, tokens in selected_domains.items():
    print(f"{domain}: {tokens/selected_tokens:.2%}")


CC: 79.84%
Books: 0.21%
Stack: 14.39%
Pes20: 2.45%
Reddit: 3.12%


# Red Pyaja mixture

CC: 67%
C4: 15%
Code: 4.5%
Wikipedia: 4.5%
Arxiv: 2.5%
Stack Exchange: 2.0%